### ISSUE TITLE
Chart: Header Component

### ISSUE THREAD

**Issue by timmyc** [woocommerce/wc-admin#361]
Child issue of #164
Designs: p6riRB-3jq-p2
[Invision](https://automattic.invisionapp.com/d/main#/projects/prototypes/14672444)

Somewhat dependent upon #359 ( but not blocked )

I'm opting to break out this issue from the parent D3 issue so we can get some more folks working on aspects of the charts to spread the 📈 wisdom across the squad.

Currently the chart on the Revenue Page is being wrapped in a `<Card />` component to get a proper wrapper/white background - but this logic should be moved to the base `<Chart />` component since it will be needed across all reports.

This task involves:

- [x] Adding requisite components or css to have the Chart component not be wrapped by a `Card`
- [x] Ensure the chart title prop is displayed properly alongside legends as outlined in the designs.
- [x] Allow Chart to be given a prop to render an Interval Selector ( see #359 ). 
- [ ] ~~Support for Ellipsis menu in chart header, akin to what exists in `<Card />`~~ Tracking in #409
- [x] Add a chart switcher ( bar <-> line ) that lives in the header, and is enabled via a prop

Likely best to split these up into smaller PRs for ease of review 😄

**Comment by timmyc** [comment]
Closing this out as remaining item is being tracked in another issue.
